### PR TITLE
Fixed exceptions being incorrectly imported

### PR DESCRIPTION
**Pull request by true-zero**
The exceptions were being imported incorrectly making it impossible to catch any raised exception

### CODE REVIEW

File: cloudscraper/__init__.py [hunk 1]
```diff
@@ -46,7 +46,7 @@
 
 # Add exceptions path
 sys.path.append(os.path.join(os.path.dirname(__file__), 'exceptions'))
-import cloudflare_exceptions  # noqa: E402
+from .exceptions.cloudflare_exceptions import * # noqa: E402
```
Comment (oczkers) [inline cloudscraper/__init__.py]: Don't You think it's better to explicitly name imports?

**Comment by VeNoMouS** [comment]
```
cloudscraper/__init__.py:49:1: F403 'from .exceptions.cloudflare_exceptions import *' used; unable to detect undefined names
cloudscraper/__init__.py:49:48: E261 at least two spaces before inline comment
cloudscraper/__init__.py:220:23: F405 'Cloudflare_Loop_Protection' may be undefined, or defined from star imports: .exceptions.cloudflare_exceptions
cloudscraper/__init__.py:303:19: F405 'Cloudflare_Block' may be undefined, or defined from star imports: .exceptions.cloudflare_exceptions
cloudscraper/__init__.py:325:19: F405 'Cloudflare_Error_IUAM' may be undefined, or defined from star imports: .exceptions.cloudflare_exceptions
cloudscraper/__init__.py:336:19: F405 'Cloudflare_Error_IUAM' may be undefined, or defined from star imports: .exceptions.cloudflare_exceptions
cloudscraper/__init__.py:364:19: F405 'Cloudflare_Error_reCaptcha' may be undefined, or defined from star imports: .exceptions.cloudflare_exceptions
cloudscraper/__init__.py:411:23: F405 'Cloudflare_reCaptcha_Provider' may be undefined, or defined from star imports: .exceptions.cloudflare_exceptions
cloudscraper/__init__.py:447:27: F405 'Cloudflare_Error_IUAM' may be undefined, or defined from star imports: .exceptions.cloudflare_exceptions
cloudscraper/__init__.py:586:19: F405 'Cloudflare_Error_IUAM' may be undefined, or defined from star imports: .exceptions.cloudflare_exceptions
```

File: cloudscraper/__init__.py (hunk 1 above)
Comment (VeNoMouS) [reply cloudscraper/__init__.py]: If you're doing relative importing , you might as well remove the sys.path appending for exceptions.

What do we prefer, full name defined objects for the exception... or just the exception name itself?

Comment (VeNoMouS) [reply cloudscraper/__init__.py]: If you're doing relative importing , you might as well  move the sys.path to the exceptions `__init__.py` and remove the noqa E402 from the exceptions import.

So I want to raise (get it ... raise, lol ... jks) this to you guys, as users... What do we prefer, full name defined objects for the exception... or just the exception name itself?

ie 

```python
raise cloudflare_exceptions.someError()
```

or 

```python
raise someError()
```

it gets a little messy when we start putting in other exception classes unrelated to the name with the same error?

**Comment by oczkers** [comment]
+1 for relative imports, of course it's just my opinion but whatever You decide please add one one or two base classes so we can catch everything easly, for example:
```python
try:
    some_code()
except cloudscraper._exceptions.CloudscraperException:
    some_other_code()
```

instead of 
```python
try:
    some_code()
except (cloudscraper.cloudflare_exceptions.Cloudflare_Loop_Protection, cloudscraper.cloudflare_exceptions.Cloudflare_Block, ...):
    some_other_code()
```



```python
class CloudflareException(Exception):
    """
    Base class for all Cloudflare exception.
    """

class LoopProtection(CloudflareException):
     """
     Raise error for recursive depth protection
     """
```

Exceptions are classes after all so why using underscores instead of CapWords?

**Comment by VeNoMouS** [comment]
based on that, what do we think of the following then?

```python
# ------------------------------------------------------------------------------- #


class CloudflareException(Exception):
    """
    Base class for all Cloudflare exception.
    """
    pass


class LoopProtection(CloudflareException):
    """
    Raise error for recursive depth protection
    """
    pass


class Blocked(CloudflareException):
    """
    Raise error for Cloudflare 1020 block
    """
    pass


class iuamError(CloudflareException):
    """
    Raise error for problem extracting IUAM paramters from Cloudflare payload
    """
    pass


class reCaptchaError(CloudflareException):
    """
    Raise error for problem extracting reCaptcha paramters from Cloudflare payload
    """
    pass


class reCaptchaProvider(CloudflareException):
    """
    Raise error for reCaptcha from Cloudflare, no provider loaded.
    """
    pass
```

```
# ------------------------------------------------------------------------------- #


class reCatpchaException(Exception):
    """
    Base class for all Cloudflare exception.
    """
    pass


class ServiceUnavailable(reCatpchaException):
    """
    Raise error for external services that cannot be reached
    """
    pass


class APIError(reCatpchaException):
    """
    Raise error for error from API response.
    """
    pass


class AccountError(reCatpchaException):
    """
    Raise error for reCaptcha provider account problem.
    """
    pass


class Timeout(reCatpchaException):
    """
    Raise error for reCaptcha provider taking too long.
    """
    pass


class BadParameter(reCatpchaException):
    """
    Raise error for bad or missing Parameter.
    """
    pass


class BadJobID(reCatpchaException):
    """
    Raise error for invalid job id.
    """
    pass


class ReportError(reCatpchaException):
    """
    Raise error for reCaptcha provider unable to report bad solve.
    """
    pass


class ImportError(reCatpchaException):
    """
    Raise error for reCaptcha, cannot import a module.
    """
    pass
```

**Comment by oczkers** [comment]
That's great :-)

```python
class ImportError(reCatpchaException):
```
How about raising default ImportError instead or inheriting from this if we really need another class? 
If i'm not mistaken that's problem with missing library so not cloudscraper/cloudflare fault. It might be little bit confusing if user is already catching reCatpchaException and for example accidently removed some library.

```python
class reCatpchaException(Exception):
```
Those recaptcha problems are "generated" by cloudflare so maybe inherit from CloudflareException instead Exception?

**Comment by VeNoMouS** [comment]
```
Those recaptcha problems are "generated" by cloudflare so maybe inherit from CloudflareException instead Exception?
```

well.. technically they don't... because that is an ABC class which uses 3rd party solvers... so its handling outside of the whole cloudflare.... while we do post back to Cloudflare, the actual process itself is handled outside of that area, hence the separation, they are all individual modules for each reCaptcha solving provider.

**Comment by oczkers** [comment]
> the actual process itself is handled outside of that area, hence the separation, they are all individual modules for each reCaptcha solving provider.

Of course, my understanding is that's consequence of iteracting with cloudflare/cloudscraper, there is no chance to get it without having fun with cloudflare. Anyway this is great change and looks very nice to me.

**Comment by VeNoMouS** [comment]
Ok, check the current dev branch 1.2.29, ive factored in some of these ideas.

**Comment by oczkers** [comment]
Looks great to me :+1:

**Comment by VeNoMouS** [comment]
Closing this off as discussed changes are in place in 1.2.30

**Pull request closed without merging**